app/src/main/java/blockeq/com/stellarwallet/activities/BaseActivity.kt

```diff
@@ -17,7 +18,7 @@ abstract class BaseActivity : AppCompatActivity() {
             if (!WalletApplication.localStore.encryptedPhrase.isNullOrEmpty() && !WalletApplication.localStore.stellarAccountId.isNullOrEmpty()) {
                 launchPINView(PinType.LOGIN, "", "", null, true)
             } else {
-                WalletApplication.localStore.clearUserData()
+                AccountUtils.wipe(this, WalletApplication.userSession.pin)
```

Comment:
do we have the pin at this point?

Reply:
Oh actually, we never have it at this point. Which means we can't clear the alias

Reply:
But in the other situation in PinActivity, we can clear it there

Reply:
I would not make the wiping depending on the pin.
To clear the keystore I would get aliases() and delete one by one.
```
   List<String> list =  keyStore.aliases()
   for(String alias : list ) {
     keyStore.deleteEntry(alias)
  }
```

Reply:
Okay that sounds good